### PR TITLE
Add type hints to specialized convolution modules.

### PR DESCRIPTION
**Pull request by jondeaton**

### CODE REVIEW

File: equinox/nn/conv.py [hunk 1]
```diff
@@ -221,16 +221,16 @@ class Conv2d(Conv):
...
-        stride=(1, 1),
-        padding=(0, 0),
-        dilation=(1, 1),
-        groups=1,
-        use_bias=True,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: Union[int, Sequence[int]],
+        stride: Union[int, Sequence[int]] = (1, 1),
+        padding: Union[int, Sequence[int]] = (0, 0),
```
Comment (patrick-kidger) [inline equinox/nn/conv.py:228]: And I think this can be a `Sequence[Tuple[int, int]]`.

**Review by patrick-kidger — COMMENTED** [review]
Thanks for contributing! I've left a couple of nitty comments. These annotations would be great to have.

File: equinox/nn/conv.py [hunk 2]
```diff
@@ -189,16 +189,16 @@ class Conv1d(Conv):
...
-        out_channels,
-        kernel_size,
-        stride=1,
-        padding=0,
-        dilation=1,
-        groups=1,
-        use_bias=True,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
```
[the remark this answers is not in the document]
Comment (jondeaton) [reply equinox/nn/conv.py]: Good point- fixed here and also on `ConvTranspose1d`.

File: equinox/nn/conv.py (hunk 1 above)
Comment (jondeaton) [reply equinox/nn/conv.py:228]: Are you sure about this? It seems to me the type can definitely be `Tuple[int, int]` but thats already covered by `Sequence[int]` (unless maybe we want to include `Tuple[int, int]` it to give the hint that the sequence must be of length 2?). Also the the documentation on `Conv` says "All of `kernel_size`, `stride`, `padding` and `dilation` can be either an integer or a sequence of integers" so I'm not sure `Sequence[Tuple[int, int]]` is valid?

Comment (patrick-kidger) [reply equinox/nn/conv.py:228]: This is used to indicate asymmetric padding.

From the docs: "`padding` can alternatively also be a sequence of 2-element tuples, each representing the padding to apply before and after each spatial dimension."

Comment (jondeaton) [reply equinox/nn/conv.py]: Thanks for pointing this out– fixed here and also for transpose convolutions.

**Comment by patrick-kidger** [comment]
LGTM! Thanks for contributing.

**Pull request merged**